private-templates-service/client/src/store/currentTemplate/reducers.ts

```diff
@@ -64,7 +69,32 @@ export function currentTemplateReducer(state = initialState, action: CurrentTemp
         sampleDataJSON: action.sampleDataJSON,
         isFetching: false
       };
+    case GET_TEMPLATE:
+      return {
+        ...state,
+        isFetching: true,
+        templateID: action.templateID,
+        template: undefined,
+        templateJSON: undefined,
+        templateName: undefined,
+        sampleDataJSON: undefined,
+      }
+    case GET_TEMPLATE_SUCCESS:
+      return {
+        ...state,
+        template: action.template,
+        templateJSON: action.templateJSON,
+        templateName: action.templateName,
+        sampleDataJSON: action.sampleDataJSON,
+        isFetching: false,
```

Comment:
for better readability, can state be assigned in the same order for each action type case